paymentsheet/src/main/java/com/stripe/android/customersheet/data/injection/CustomerAdapterDataSourceModule.kt

```diff
@@ -0,0 +1,37 @@
+package com.stripe.android.customersheet.data.injection
+
+import com.stripe.android.customersheet.CustomerAdapter
+import com.stripe.android.customersheet.ExperimentalCustomerSheetApi
+import com.stripe.android.customersheet.data.CustomerAdapterDataSource
+import com.stripe.android.customersheet.data.CustomerSheetIntentDataSource
+import com.stripe.android.customersheet.data.CustomerSheetPaymentMethodDataSource
+import com.stripe.android.customersheet.data.CustomerSheetSavedSelectionDataSource
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+
+@OptIn(ExperimentalCustomerSheetApi::class)
+@Module
+internal interface CustomerAdapterDataSourceModule {
+    @Binds
+    fun bindsCustomerSheetPaymentMethodDataSource(
+        impl: CustomerAdapterDataSource
```

Comment:
This will give us multiple instances of CustomerAdapterDataSource and CustomerAdapter. In practice this might be fine?

Reply:
Should we make `CustomerAdapterDataSource` a singleton?

Reply:
I thought it was a singleton already but it does not appear so. We should make it a singleton